### PR TITLE
[LI-HOTFIX] Improve passthrough performance in Log.append

### PR DESCRIPTION
**Pull request by lmr3796**
TICKET = LIKAFKA-9737
LI_DESCRIPTION = Broker-side changes for pass-through solution. This is the porting for commit 0d9c527

EXIT_CRITERIA = MANUAL ["If we no longer need pass-through support in Kafka. Only known user is KMM for venice; BrooklinMM has not enabled this everywhere."]

### CODE REVIEW

**Comment by lmr3796** [comment]
This is the porting for commit 0d9c52774c .

Please see my in line comments below for the difference

File: core/src/main/scala/kafka/log/Log.scala [hunk 1]
```diff
@@ -759,6 +759,87 @@ class Log(@volatile private var _dir: File,
...
+      var maxTimestamp = info1.maxTimestamp
+      var offsetofMaxTimestamp = info1.offsetOfMaxTimestamp
+      if (info2.maxTimestamp > info1.maxTimestamp) {
+        maxTimestamp = info2.maxTimestamp
+        offsetofMaxTimestamp = info2.offsetOfMaxTimestamp
+      }
+      val combinedRecordConversionStats = info1.recordConversionStats
+      combinedRecordConversionStats.add(info2.recordConversionStats)
+
+      val lastLeaderEpoch: Option[Int] = (info1.lastLeaderEpoch, info2.lastLeaderEpoch) match {
```
Comment (lmr3796) [inline core/src/main/scala/kafka/log/Log.scala]: This is the key difference from the original commit

In apache#9382, `LogAppendInfo` takes an additional `lastLeaderEpoch`, so in the batch combination logic I take the max from all of them.

Comment (gitlw) [reply core/src/main/scala/kafka/log/Log.scala]: The doc says 
* @param lastLeaderEpoch The partition leader epoch corresponding to the last offset, if available.
so I think it makes sense to simply pass along the result based on the later item, i.e. info2.lastLeaderEpoch.

**Review by gitlw — APPROVED** [review]
Thanks for this change!

**Pull request merged**